sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -138,16 +138,17 @@ cdef np.ndarray vector_to_nd_array(vector_DITYPE_t * vect_ptr):
 
 
 cdef np.ndarray[object, ndim=1] coerce_vectors_to_nd_arrays(
-    shared_ptr[vector_vector_DITYPE_t] vecs
+    shared_ptr[vector_vector_DITYPE_t] vects_ptr
 ):
     """Coerce a std::vector of std::vector to a ndarray of ndarray."""
     cdef:
-        ITYPE_t n = deref(vecs).size()
+        vector_vector_DITYPE_t vects = deref(vects_ptr)
```

Comment:
Same comment here: I think we hardly can prevent a copy if we want to define a local `vects`.